### PR TITLE
Stop matching pre-release suffixed next version

### PR DESCRIPTION
**Pull request by norio-nomura**
Fix https://bugs.swift.org/browse/SR-787

### CODE REVIEW

**Comment by norio-nomura** [comment]
Updated.

**Comment by mxcl** [comment]
Thanks, though we need a test to prevent regression.

**Comment by norio-nomura** [comment]
Rebased and added test.

**Comment by mxcl** [comment]
@swift-ci Please Test

**Comment by kostiakoval** [comment]
@swift-ci is available for SwiftPM now? 😲 WoW

**Comment by aciidb0mb3r** [comment]
woah, can others trigger it or only admins?

**Comment by mxcl** [comment]
Only “Apple Members”, the fear was allowing anyone to trigger remote code execution.

**Comment by aciidb0mb3r** [comment]
I understand

**Comment by mxcl** [comment]
But, seems broken. I have tested locally and for some reason this patch causes a failure in ManifestTests on Linux—but not Darwin.

**Comment by mxcl** [comment]
```
Test Case 'ManifestTests.testManifestLoading' started.
/home/mxcl/src/swiftpm/Tests/ManifestParser/ManifestTests.swift:40: error: ManifestTests.testManifestLoading : XCTAssertEqual failed: ("[PackageDescription.Package.Dependency]") is not equal to ("[PackageDescription.Package.Dependency]") - 
Test Case 'ManifestTests.testManifestLoading' failed (0.202 seconds).
```

**Comment by modocache** [comment]
Could it be that `XCTAssertEqual` itself behaves differently in swift-corelibs-xctest? 🤔

What are the types of the two inputs?

**Comment by shahmishal** [comment]
@swift-ci Please Test

**Comment by shahmishal** [comment]
Pull request testing system for swiftpm is working now. 
Sorry for the trouble.

**Comment by norio-nomura** [comment]
I hope this will be in next SNAPSHOT…

**Comment by norio-nomura** [comment]
Oh, new SNAPSHOT has been released without this. :cry: 
What can I do for merging this?

**Comment by norio-nomura** [comment]
Rebased.

**Comment by mxcl** [comment]
@norio-nomura sorry a test was failing, so I didn't merge. Let me try again.

**Comment by mxcl** [comment]
Yes, I still get a failure on Linux, but I do not on Darwin, and I do not get this failure on Linux for master branch.

```
Test Case 'ManifestTests.testManifestLoading' started.
/home/mxcl/src/swiftpm/Tests/ManifestParser/ManifestTests.swift:40: error: ManifestTests.testManifestLoading : XCTAssertEqual failed: ("[PackageDescription.Package.Dependency]") is not equal to ("[PackageDescription.Package.Dependency]") - 
Test Case 'ManifestTests.testManifestLoading' failed (0.225 seconds).
Executed 1 test, with 1 failure (0 unexpected) in 0.225 (0.225) seconds
```

**Comment by aciidb0mb3r** [comment]
weird, it passes for me on Ubuntu 15.05
```
Test Case 'ManifestTests.testManifestLoading' started.
Test Case 'ManifestTests.testManifestLoading' passed (0.17 seconds).
Executed 1 test, with 0 failures (0 unexpected) in 0.17 (0.171) seconds
```

```
$ swiftc -v
Swift version 3.0-dev (LLVM a7663bb722, Clang 4ca3c7fa28, Swift 1c2f40e246)
Target: x86_64-unknown-linux-gnu
```
--
@norio-nomura unrelated but you also probably need to add `PackageDescriptiontest.PackageTests()` in `LinuxMain.swift`

**Comment by norio-nomura** [comment]
Thanks all. :pray: 
@aciidb0mb3r I added that!

**Comment by mxcl** [comment]
@swift-ci Please Test

**Comment by norio-nomura** [comment]
Yay! Test passed! :raised_hands:

**Comment by norio-nomura** [comment]
Thanks! 🙏
